### PR TITLE
PCHR-779: Document Modal Design & New Functionality

### PR DESCRIPTION
**Pull request by kkalamarski**
Before:
before-779

After: 
779-after

Dropzone Click:
get file

### CODE REVIEW

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js [hunk 1]
```diff
@@ -187,22 +186,36 @@ define(['controllers/controllers',
...
                     $modalInstance.dismiss();
                     $scope.$broadcast('ta-spinner-hide');
                     return $q.reject();
                 });
-            }
+            };
+
+            $scope.dpOpen = function ($event, name) {
+                $event.preventDefault();
+                $event.stopPropagation();
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: I think both `stopPropagation` and `preventDefault` are not needed, given that the anchor elements from which `dpOpen` is called have the `href` attribute without a value

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js [hunk 2]
```diff
@@ -187,22 +186,35 @@ define(['controllers/controllers',
...
+                $scope.dpOpened[name] = true;
+            };
 
+            $scope.dropzoneClick = function () {
+                $timeout(function () {
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: Why the `$timeout`?

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js [hunk 3]
```diff
@@ -187,22 +186,35 @@ define(['controllers/controllers',
...
+                    document.getElementById($rootScope.prefix + 'document-files').click();
```
Comment (AkA84) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: I understand the time constraints, but this is a design flaw. AngularJS controllers should never manipulate the DOM directly, that's what the directives (through the `link` function) are for

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js (hunk 1 above)
Comment (kkalamarski) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: It's copy-paste from Tasks Dashboard. I thought the same, but when I removed that it stopped working. I decided not to investigate it right now, as we have limited time.

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js (hunk 2 above)
Comment (kkalamarski) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: Need to wait until next digest cycle.

**Comment by AkA84** [comment]
Approved

File: uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js (hunk 3 above)
Comment (kkalamarski) [inline uk.co.compucorp.civicrm.tasksassignments/js/controllers/modal/modalDocument.js]: Agree

**Pull request merged**